### DISCUSSION TITLE
Weekly release: 0.19.0.dev2025040100

### DISCUSSION THREAD

**Discussion by kaiyux** [NVIDIA/TensorRT-LLM#3204] [Announcements]
Hi,

The TensorRT-LLM team is pleased to announce that we have updated a weekly release `0.19.0.dev2025040100`, and pushed an [update](https://github.com/triton-inference-server/tensorrtllm_backend/pull/731) to the [Triton backend](https://github.com/triton-inference-server/tensorrtllm_backend/tree/main) this April 1, 2025. 

The `0.19.0.dev2025040100` dev release includes:

* Model Support
  - Added EXAONE-Deep support. Refer to `examples/exaone/README.md`. (#3054)
  - Added one-shot version for UserBuffer AllReduce-Normalization on FP16/BF16. (#2995)
  - Added initial EAGLE-3 implementation. (#3035)
* Features
  - Added PyTorch pipeline parallelism with attention DP support. (#3044)
  - Added support for enabling MTP with CUDA graph padding. (#3096)
  - Added request BW metric measurement for `disaggServerBenchmark`. (#3070)
  - Updated logits bitmask kernel to v3 (#3009)
  - Supported AutoDeploy as a backend for `trtllm-bench` command (#3041)
  - Enabled CUDA graphs when attention DP was used and active requests on different GPUs were uneven (#3010)
  - Added iteration log support for `trtllm-bench` (#3039)
* API
  - Moved `BuildConfig` arguments to `LlmArgs`. (#3036)
  - Removed speculative decoding parameters from stateful decoders (#3024)
* Bug fixes
  - Fixed the missing parameter for the WeightOnlyQuantRowLinear module (#2768)
  - Fixed the single-node quitting issue on SLURM (#3140)
  - Fixed early exit in CMake when `find_library()` did not find any library (#3113)
  - Fixed a hang in MGMN cases when using the `trtllm-llmapi-launch` command (#3119)
  - Fixed `gpus_per_node` issue in `trtllm-bench` when `world_size` is less than `device_count` (#3007)
  - Fixed misaligned fuse messages across different processes (#3067)
  - Fixed the issue when `cp_size` is greater than `kvHeadNum` (#3002)
  - Fixed incorrect `draft_token_nums` for dummy requests during torch compilation with MTP (#3053)
  - Fixed logits dtype in assertion (#3038)

The cut-off commit to this release is 75495730bca7a4c94feeed29b17014594f1a1598. The code changes can be seen here: https://github.com/NVIDIA/TensorRT-LLM/compare/c2ffce7...7549573.

Thanks,
The TensorRT-LLM Engineering Team
